hibernate-core/src/main/java/org/hibernate/query/spi/AbstractCommonQueryContract.java

```diff
@@ -906,7 +907,7 @@ public CommonQueryContract setParameter(int position, Object value) {
 		final QueryParameter<Object> param = binding.getQueryParameter();
 		if ( param.allowsMultiValuedBinding() ) {
 			final BindableType<?> hibernateType = param.getHibernateType();
-			if ( hibernateType == null || isInstance( hibernateType, value ) ) {
+			if ( hibernateType == null || hibernateType instanceof NullSqmExpressible || isInstance( hibernateType, value ) ) {
```

Comment:
@dreab8 Why did you make this change for positional parameters, but not for named parameters?

Reply:
Hi @gavinking, 

a big oversight :( I'm going to open a Jira and fix it.

Reply:
thanks for spotting it

Reply:
Excellent, cheers mate 👍